signup/signup.js

```diff
@@ -0,0 +1,221 @@
+const allInputs = document.querySelectorAll('input');
+const emailInput = document.querySelector('#e-mail');
+const nullErr = document.querySelector('.null-err');
+const notAnEmailErr = document.querySelector('.not-an-email-err');
+
+// focus가 될때 blue, blur가 될때 원래의 테두리를 가지는 코드(css스위치)
+allInputs.forEach((input) => {
+  input.addEventListener('focus', () => input.classList.add('focused-border'));
+  input.addEventListener('blur', () => {
+    input.classList.remove('focused-border');
+  });
+});
+
+// 이메일 유효성 검사 코드 - 반환값은 3개이다. true/false/이메일확인용변수
+let notAnEmail = 2; // <- 반환값 확인용 변수 선언. -> let? const?
+function isEmailValid(email) {
+  // 이메일 정규표현식
+  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+  // 입력값이 빈 값인지 확인하고, 이메일 유효성 검사
+  if (email.trim() === '') {
+    return false;
+  } else if (!emailRegex.test(email)) {
+    return notAnEmail;
+  }
+  return true;
+}
+
+// focus가 될때 실행될 코드
+emailInput.addEventListener('focus', function () {
+  emailInput.classList.add('focused-border'); //focus 기본동작
+
+  //빈값 에러가 존재한다면 실행될 코드
+  if (nullErr.style.display === 'block') {
+    emailInput.addEventListener('keyup', function () {
+      emailInput.classList.remove('err-border');
+      nullErr.style.display = 'none';
+    });
+  }
+  //유효성x 에러가 존재한다면 실행될 코드
+  if (notAnEmailErr.style.display === 'block') {
+    // 만약 이메일 유효성 에러div가 존재한다면
+    emailInput.addEventListener('keyup', function (e) {
+      //사용자의 값을 받는 keyup실행
+      const emailValue = e.target.value;
+      if (isEmailValid(emailValue) === notAnEmail) {
+        //이메일값이 유효 x 그냥 에러 그대로
+        //nothing
+      } else if (isEmailValid(emailValue)) {
+        //만약 이메일값이 유효하다면 그즉시 에러와 메시지 삭제.
+        emailInput.classList.remove('err-border');
+        notAnEmailErr.style.display = 'none';
+      } else {
+        //nothing
+      }
```

Comment:
```suggestion
      } 
```
아무 작업이 없다면 불필요한 코드인것 같아요!